### PR TITLE
add an entity example to the DotNetIsolated smoke test project.

### PR DESCRIPTION
**Pull request by sebastianburckhardt**
Adds a simple entity example to the DotNetIsolated smoke test.

### Pull request checklist

goes into feature branch. Checklist items are tracked by the feature branch.

### CODE REVIEW

File: test/SmokeTests/OOProcSmokeTests/DotNetIsolated/Counter.cs [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+using Microsoft.DurableTask.Entities;
+using Microsoft.Extensions.Logging;
+
+namespace DotNetIsolated;
+
+
+/// <summary>
+/// A simple counter, demonstrating entity use.
+/// </summary>
+[DataContract]
```
Comment (jviau) [inline test/SmokeTests/OOProcSmokeTests/DotNetIsolated/Counter.cs:21]: Data attributes are not necessary. They might even be misleading; STJ does not support them by default. You have to implement a completely custom `STJ.DataContractResolver`. Would be better to remove and let customers figure out serialization through existing STJ examples/docs.

File: test/SmokeTests/OOProcSmokeTests/DotNetIsolated/Counter.cs [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+            int currentValue = response.ReadStateAs<Counter>()!.CurrentValue;
+            var httpResponse = request.CreateResponse(System.Net.HttpStatusCode.OK);
+            httpResponse.Headers.Add("Content-Type", "text/plain; charset=utf-8");
+            httpResponse.WriteString($"{currentValue}\n");
+            return httpResponse;
+        }
+    }
+
+    [Function(nameof(DeleteCounter))]
+    public static async Task<HttpResponseData> DeleteCounter(
```
Comment (jviau) [inline test/SmokeTests/OOProcSmokeTests/DotNetIsolated/Counter.cs:126]: Can you add a comment explaining how delete is an implicit operation.

``` CSharp
// All entities have a "delete" operation built into them.
```

File: test/SmokeTests/OOProcSmokeTests/DotNetIsolated/Counter.cs (hunk 1 above)
Comment (sebastianburckhardt) [reply test/SmokeTests/OOProcSmokeTests/DotNetIsolated/Counter.cs:21]: The goal was to clarify to the reader that this class is going to be serialized when saving the entity state.
Its generally something I like to make very clear, even if attributes are not technically required.

What is the "STJ way" to accomplish this? Also, what is STJ?

Comment (sebastianburckhardt) [reply test/SmokeTests/OOProcSmokeTests/DotNetIsolated/Counter.cs]: (For now I have just removed this)

Comment (davidmrdavid) [reply test/SmokeTests/OOProcSmokeTests/DotNetIsolated/Counter.cs]: I think STJ is `System.Text.JSON`

File: test/SmokeTests/OOProcSmokeTests/DotNetIsolated/Counter.cs (hunk 2 above)
Comment (davidmrdavid) [reply test/SmokeTests/OOProcSmokeTests/DotNetIsolated/Counter.cs:123]: does that mean 'delete' is a reserved operation name? May be worth calling that out if so

Comment (sebastianburckhardt) [reply test/SmokeTests/OOProcSmokeTests/DotNetIsolated/Counter.cs:123]: It's not exactly 'reserved' it just has a default implemntation already defined.

**Pull request merged**